File: medium/cpp/848.cpp
// https://leetcode.com/problems/shifting-letters/

class Solution
{
public:
    std::string shiftingLetters(std::string s, std::vector<int>& shifts)
    {
        const auto n = static_cast<int>(std::size(s));

        for (auto i = n, acc = 0; i--; )
        {
            s[i] = (s[i] - 'a' + (acc += shifts[i])) % 26 + 'a';
            acc %= 26;
        }

        return s;
    }
};

File: medium/python/848.py
# https://leetcode.com/problems/shifting-letters/

class Solution:
    def shiftingLetters(self, s: str, shifts: List[int]) -> str:
        acc = 0
        res = []

        for i in range(len(s) - 1, -1, -1):
            res.append(chr(ord('a') + (ord(s[i]) - ord('a') + shifts[i] + acc) % 26))
            acc = (acc + shifts[i]) % 26

        return ''.join(res)[::-1]